src/docfx/apply-template/ApplyTemplate.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class ApplyTemplate
+    {
+        public static bool Run(CommandLineOptions options)
+        {
+            using var errors = new ErrorWriter(options.Log);
+
+            if (string.IsNullOrEmpty(options.Template))
+            {
+                errors.Add(Errors.ApplyTemplate.TemplateNotSpecified());
+                return errors.HasError;
+            }
+
+            var directory = options.Input ?? ".";
+
+            var (templateEngineRunner, trustedDomains) = Prepare(errors, options.Template, options.Locale ?? "en-us");
+
+            Parallel.ForEach(
+                Directory.GetFiles(directory, "*.*", SearchOption.AllDirectories),
+                filePath =>
+                {
+                    ApplyTemplates(errors, filePath, directory, options.Output ?? directory, templateEngineRunner, trustedDomains, options.Locale);
```

Comment:
nit: Rename to `ApplyTemplateCore`, better to be consistent with the class name?

Reply:
Done.